Kernel/Tasks/Process.h

```diff
@@ -605,6 +614,22 @@ class Process final
     ErrorOr<void> validate_mmap_prot(int prot, bool map_stack, bool map_anonymous, Memory::Region const* region = nullptr) const;
     ErrorOr<void> validate_inode_mmap_prot(int prot, bool description_readable, bool description_writable, bool map_shared) const;
 
+    template<size_t Size>
+    static ErrorOr<void> get_syscall_string_argument_into_static_char_buffer(Syscall::StringArgument const& argument, FixedCharBuffer<Size>& buffer)
+    {
+        // NOTE: If the string is too much big for the FixedCharBuffer,
+        // we return E2BIG error here.
+        return try_copy_string_from_user_into_static_char_buffer<Size>(reinterpret_cast<FlatPtr>(argument.characters), buffer, argument.length);
```

Comment:
Doesn't this cast remove the `Userspace<>` from the syscall argument? This seems less than ideal. We shouldn't be reinterpret casting when it loses valuable data

Reply:
Is it the case? I was sure that passing a FlatPtr is needed to actually get the `Userspace<>` wrapper in the StdLib function I created...